src/main/java/com/fasterxml/jackson/databind/deser/SettableAnyProperty.java

```diff
@@ -437,4 +481,99 @@ public SettableAnyProperty withValueDeserializer(JsonDeserializer<Object> deser)
+        implements java.io.Serializable
+    {
+        private static final long serialVersionUID = 1L;
+
+        protected final ValueInstantiator _valueInstantiator;
+
+        protected final int _parameterIndex;
+
+        public MapParameterAnyProperty(BeanProperty property, AnnotatedMember field, JavaType valueType,
+                KeyDeserializer keyDeser, JsonDeserializer<Object> valueDeser, TypeDeserializer typeDeser,
+                ValueInstantiator inst, int parameterIndex)
+        {
+            super(property, field, valueType, keyDeser, valueDeser, typeDeser);
+            _valueInstantiator = Objects.requireNonNull(inst, "ValueInstantiator for MapParameterAnyProperty cannot be `null`");
+            _parameterIndex = parameterIndex;
+        }
+
+        @Override
+        public SettableAnyProperty withValueDeserializer(JsonDeserializer<Object> deser)
+        {
+            return new MapParameterAnyProperty(_property, _setter, _type, _keyDeserializer, deser,
+                    _valueTypeDeserializer, _valueInstantiator, _parameterIndex);
+        }
+
+        @Override
+        protected void _set(Object instance, Object propName, Object value)
+        {
+            ((Map<Object, Object>) instance).put(propName, value);
+        }
+
+        @Override
+        public int getParameterIndex() { return _parameterIndex; }
+
+        @Override
+        public Object createParameterObject() { return new HashMap<>(); }
+
+    }
+
+    /**
+     * [databind#562] Allow @JsonAnySetter on Creator constructor
+     *
+     * @since 2.18
+     */
+    protected static class JsonNodeParameterAnyProperty extends SettableAnyProperty
+        implements java.io.Serializable
+    {
+        private static final long serialVersionUID = 1L;
+
+        protected final JsonNodeFactory _nodeFactory;
+
+        protected final int _parameterIndex;
+
+        public JsonNodeParameterAnyProperty(BeanProperty property, AnnotatedMember field, JavaType valueType,
+                JsonDeserializer<Object> valueDeser, JsonNodeFactory nodeFactory, int parameterIndex)
+        {
+            super(property, field, valueType, null, valueDeser, null);
+            _nodeFactory = nodeFactory;
+            _parameterIndex = parameterIndex;
+        }
+
+        // Let's override since this is much simpler with JsonNodes
+        @Override
+        public Object deserialize(JsonParser p, DeserializationContext ctxt)
+                throws IOException
+        {
+            return _valueDeserializer.deserialize(p, ctxt);
+        }
+
+        @Override
+        protected void _set(Object instance, Object propName, Object value)
+                throws Exception
+        {
+            ((ObjectNode) instance).set((String) propName, (JsonNode) value);
+        }
+
+        // Should not get called but...
+        @Override
+        public SettableAnyProperty withValueDeserializer(JsonDeserializer<Object> deser) { return this; }
```

Comment:
If it should not get called (comment may be copied from somewhere else), should throw an exception I think.
`UnsupportedOperationException` f.ex.

Reply:
It should not get called, but does get called. Same as `JsonNodeFieldAnyProperty`. Lemme re-check if I can change.

Reply:
Applied via https://github.com/FasterXML/jackson-databind/pull/4558/commits/0e11a15e3e2626011005d961e160333c764dcec5